cpp/src/arrow/compute/exec.h

```diff
@@ -257,6 +257,10 @@ struct ARROW_EXPORT ExecBatch {
     return result;
   }
 
+  /// \brief A check for the empty ExecBatch with no output length
+  /// and function's arguments.
+  bool IsNull() const { return length == 0 && num_values() == 0; }
```

Comment:
Under what circumstances would we have `num_values() == 0` but `length != 0`? It seems accessing the length of an ExecBatch with no values would be ill formed and we could forgo `IsValid()` in favor of just checking `num_values() == 0`

Reply:
Agree, seems `num_values() == 0` is enough.